redbot/cogs/warnings/warnings.py

```diff
@@ -422,6 +428,29 @@ def __init__(self, bot: Red):
         if user_id == ctx.author.id:
             return await ctx.send(_("You cannot remove warnings from yourself."))
 
+        custom_allowed = await self.config.guild(ctx.guild).allow_custom_reasons()
+        guild_settings = self.config.guild(ctx.guild)
+        reason_type = None
+        async with guild_settings.reasons() as registered_reasons:
+            if reason.lower() not in registered_reasons:
+                msg = _("That is not a registered reason.")
+                if custom_allowed:
+                    reason_type = {"description": reason}
+                elif (
```

Comment:
Actually stating the valid reasons, when `custom_allowed` is disallowed and `reason.lower() not in registered_reasons`  would be more user friendly (something along the lines of "Please use one of the following: ...")

Right now `if reason.lower() not in registered_reasons and not custom_allowed` this just fails silently

Reply:
Will look at this when I return home later today

Reply:
This would then require to be edited in the warn command as well. My knowledge in string formatting is limited (everyone tells me my code is to slow) that I'd rather leave that up to someone else as it also doesn't really fit in this PR in my opinion. If you really want to see this included I'll need some guidance via discord.

I'm guessing that ``reason_lower`` is a list of strings that in one way ore another can be concatinated and passed to an embed which is then sent to the user. If we exceed the embed limit with the amount of registered reasons that's going to require some logic I don't know how to code and I'm not really sure if it's worth it at that point. Leaving it with just the instructions on how to enable custom warnings seems reasonable to me.